### ISSUE TITLE
JSON serializer does not handle null values in a 2D array well

### ISSUE THREAD

**Issue by wdavies973** [libgdx/libgdx#5447]
I'm not entirely sure whether this is expected behavior or not, but I am serializing a 2D array of objects `MyObjects[][]`. The array is 30x30 AND some objects are null in it at any given time. I have implemented Json.Serializable in the `MyObject` class. I track about 10 values inside the `MyObject` class that get serialized and unserialized. The issue is, the built in LibGDX Json utility thinks that a "null" value in the `MyObjects[][]` array should attempt to instantiate the object within the array anyway. 

```java
json.setSerializer(MyObject.class, new Json.Serializer<MyObject>() {
            @Override
            public void write(Json json, MyObject object, Class knownType) {
                object.write(json);
            }

            @Override
            public MyObject read(Json json, JsonValue jsonData, Class type) {
                if(jsonData.isNull()) return null;

                // This prints "null null null null null", in other words, apparently the jsonData for this
                // class isn't null, but all of its member variables are. Also, a class is only serialized according
                // to the write method() above, so I'm not sure how jsonData.isNull() returns false here

                System.out.println("Here: "+jsonData+".");

                MapBuilding building = new MapBuilding();
                building.read(json, jsonData);
                return building;
            }
        });
```

Instead, the correct behavior should be that if `null` is found in the serialized JSON data, LibGDX should make sure the object gets deserialized as `null` when it's inserted back into the array. 

In short: values that are `null` when serialized should be `null` when deserialized. 

I will provide raw JSON and more code if necessary.

**Comment by NathanSweet** [comment]
Please show code (eg, single class with a main method) I can run to see the problem and verify a fix. With that I'll reopen the issue!
https://github.com/libgdx/libgdx/wiki/More-Info

**Comment by wdavies973** [comment]
@NathanSweet 

I am serializing the following class `Save`. The array `MapBuilding[][]` may contain buildings in it, but any spot may be null (to represent no building on that tile). 

```java
public class Save implements Json.Serializable {

    private MapBuilding[][] buildings;

    public Save() {
         buildings = new MapBuilding[30][30];
    }

    @Override
    public void write(Json json) {
        json.writeValue("buildings", buildings);
    }

    @SuppressWarnings("unchecked")
    @Override
    public void read(Json json, JsonValue jsonData) {
        buildings = json.readValue("buildings", MapBuilding[][].class, MapBuilding.class, jsonData);
    }
}```

Upon serializing the save file, I get the following [JSON](https://pastebin.com/0uup6f49). As you can see, the save file contains mostly null `MapBuilding`s, but has a few non-null spots.

However, when  deserializing using the following code:
```java
json.setSerializer(MapBuilding.class, new Json.Serializer<MapBuilding>() {
            @Override
            public void write(Json json, MapBuilding object, Class knownType) {
                object.write(json);
            }

            @Override
            public MapBuilding read(Json json, JsonValue jsonData, Class type) {
                if(jsonData.isNull()) return null;
                
               // jsonData contains: [null null null null null]

                MapBuilding building = new MapBuilding();
                building.read(json, jsonData);
                return building;
            }
        });
```
The JSON data attempts to read a null object from the `MapBuilding`. The `read()` method receives non-null `jsonData` that contains `null` for every single `MapBuilding` member variable. Instead, the `JSON` deserializer should just set `null` objects in the `MapBuilding[][]` array to null instead of creating a new `MapBuilding` and trying to set null variables to it (which causes the program to crash in the `MapBuilding` `read()` method.

Several other things:

- The same problem occurs when I don't use `json.setSerializer()...`
- `json.setUsePrototypes(false)` is set
- No other `JSON` object variables have been changed.

**Comment by NathanSweet** [comment]
There might be a bug here, but I can't tell because there isn't code I can run to see the problem.
